service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformManager.java

```diff
@@ -0,0 +1,165 @@
+package com.solace.maas.ep.event.management.agent.plugin.terraform.manager;
+
+import com.solace.maas.ep.event.management.agent.plugin.command.model.Command;
+import com.solace.maas.ep.event.management.agent.plugin.command.model.CommandRequest;
+import com.solace.maas.ep.event.management.agent.plugin.command.model.CommandResult;
+import com.solace.maas.ep.event.management.agent.plugin.command.model.JobStatus;
+import com.solace.maas.ep.event.management.agent.plugin.terraform.client.TerraformClient;
+import com.solace.maas.ep.event.management.agent.plugin.terraform.client.TerraformClientFactory;
+import com.solace.maas.ep.event.management.agent.plugin.terraform.configuration.TerraformProperties;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.MDC;
+import org.springframework.stereotype.Service;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+
+import static com.solace.maas.ep.event.management.agent.plugin.constants.RouteConstants.COMMAND_CORRELATION_ID;
+import static com.solace.maas.ep.event.management.agent.plugin.constants.RouteConstants.MESSAGING_SERVICE_ID;
+
+@Service
+@Slf4j
+public class TerraformManager {
+    private final TerraformLogProcessingService terraformLogProcessingService;
+    private final TerraformProperties terraformProperties;
+    private final TerraformClientFactory terraformClientFactory;
+
+    private final static String TF_CONFIG_FILENAME = "config.tf";
+
+    public TerraformManager(TerraformLogProcessingService terraformLogProcessingService,
+                            TerraformProperties terraformProperties, TerraformClientFactory terraformClientFactory) {
+        this.terraformLogProcessingService = terraformLogProcessingService;
+        this.terraformProperties = terraformProperties;
+        this.terraformClientFactory = terraformClientFactory;
+    }
+
+    public void execute(CommandRequest request, Command command, Map<String, String> envVars) {
+
+        MDC.put(COMMAND_CORRELATION_ID, request.getCorrelationId());
+        MDC.put(MESSAGING_SERVICE_ID, request.getServiceId());
+        String traceId = MDC.get("traceId");
+        String spanId = MDC.get("spanId");
+
+        log.debug("Executing command {} for ms {} correlationId {} context {}", command.getCommand(), request.getServiceId(),
```

Comment:
please change `ms` to some generic name like `service` since it won't work for schemaRegsitry configPush scenarios